besu/src/main/java/org/hyperledger/besu/RunnerBuilder.java

```diff
@@ -590,7 +592,8 @@ private FilterManager createFilterManager(
       final PrivacyParameters privacyParameters,
       final JsonRpcConfiguration jsonRpcConfiguration,
       final WebSocketConfiguration webSocketConfiguration,
-      final MetricsConfiguration metricsConfiguration) {
+      final MetricsConfiguration metricsConfiguration,
+      final Optional<NatService> natService) {
```

Comment:
This is a general comment outside of the scope of this PR but why do we have an `Optional<NatService>` and also a `NatMethod.NONE`? Is there an important difference between them or can we consolidate the concepts into one of the two?

Reply:
yes indeed, it is planned to delete the optional for the NatService in a future PR. I prefer to have a PR just for that because it changes a lot of file (the original implementation used an optional). I can do that in the next PR if you want.

Reply:
Perfect! As long as we do that in one of the PRs before this all gets merged then that's fine.